meta-lmp-base/classes/lmp-staging.bbclass

```diff
@@ -38,3 +38,38 @@ def lmp_sstate_checkhashes(sq_data, d, **kwargs):
     if mirrors:
         bb.plain("SState mirrors: %s" % mirrors)
     return sstate_checkhashes(sq_data, d, **kwargs)
+
+# don't check images
+LMPSTAGING_DEPLOYED_CHECK_SKIP ?= "${@d.getVar('PN') if bb.data.inherits_class('image', d) else ''}"
+
+# handler to warn when there are 'do_deploy' dependencies without setting the DEPENDS
+addhandler check_deployed_depends
+check_deployed_depends[eventmask] = "bb.build.TaskSucceeded"
+python check_deployed_depends() {
+    d = e.data
+
+    taskname = d.getVar('BB_RUNTASK')
+    pn = d.getVar('PN')
+    pn = [pn, '%s:%s' % (pn, taskname)]
+    excludes = (d.getVar('LMPSTAGING_DEPLOYED_CHECK_SKIP') or '').split()
+    for s in excludes:
+        if s in pn:
+            #bb.warn("skip '%s' deployed dependencies check" % s)
+            bb.debug(1, "skip '%s' deployed dependencies check" % s)
+            return
+
+    depends = (d.getVarFlag(taskname, 'depends') or '').split()
+    # remove duplicates
+    depends = [*set(depends)]
+    for depend in depends:
+        recipe, task = depend.split(':')
+        if task == 'do_deploy' and recipe not in d.getVar('DEPENDS'):
+            bb.error("Task '%s' depends on '%s' but '%s' is not in DEPENDS" % (taskname, depend, recipe))
```

Comment:
I think it is better for us to warn by default instead of calling an error here, as this might have side effects in customer related recipes and cause certain builds to fail.

While we do want to have SPDX in a correct form, we shouldn't just yet enforce this on every recipe (we can have as a warn first and on another release move to error, for example).

Reply:
I put a new commit [25a255e](https://github.com/foundriesio/meta-lmp/pull/1109/commits/25a255ec5211dff0c185ddcfd8047959b256be23) wih this that is simpler to reference later.